bugsnag-android-core/src/main/java/com/bugsnag/android/RemoteConfig.kt

```diff
@@ -0,0 +1,109 @@
+
+    internal companion object : JsonReadable<RemoteConfig?> {
+        internal const val KEY_CONFIGURATION_TAG = "configurationTag"
+        internal const val KEY_CONFIG_EXPIRY = "configurationExpiry"
+        internal const val KEY_DISCARD_RULES = "discardRules"
+        internal const val DISCARD_RULE_ALL = "ALL"
+        internal const val DISCARD_RULE_ALL_HANDLED = "ALL_HANDLED"
+        internal const val MATCH_TYPE = "match_type"
+
+        override fun fromReader(reader: JsonReader): RemoteConfig? {
+            var remoteConfig: RemoteConfig? = null
+            with(reader) {
+                beginObject()
+                var configurationTag: String? = null
+                var configExpiry: Date? = null
+                var discardRules: List<DiscardRule> = emptyList()
+
+                while (hasNext()) {
+                    val key = nextName()
+                    when (key) {
+                        KEY_CONFIGURATION_TAG -> configurationTag = nextString()
+                        KEY_CONFIG_EXPIRY -> configExpiry = DateUtils.fromIso8601(nextString())
+                        KEY_DISCARD_RULES -> {
+                            discardRules = parseDiscardRules(reader)
+                        }
+                    }
+                }
+                if (configurationTag != null && configExpiry != null) {
+                    remoteConfig = RemoteConfig(configurationTag, configExpiry, discardRules)
+                }
+                endObject()
+            }
+            return remoteConfig
+        }
+
+        private fun parseDiscardRules(reader: JsonReader): List<DiscardRule> {
+            val rules = ArrayList<DiscardRule>()
+            var rule: DiscardRule? = null
+            with(reader) {
+                beginObject()
+                beginArray()
+                while (hasNext()) {
+                    beginObject()
+                    while (hasNext()) {
+                        val key = nextName()
+                        when (key) {
+                            MATCH_TYPE -> {
+                                when (nextString()) {
+                                    DISCARD_RULE_ALL -> rule = DiscardRule.All()
+                                    DISCARD_RULE_ALL_HANDLED -> rule = DiscardRule.AllHandled()
+                                }
+                            }
+                        }
+                        endObject()
+                    }
+                    endObject()
+                }
+                endArray()
+                endObject()
+            }
+            rule?.let { rules.add(it) }
+            return rules
+        }
+    }
+}
+
+internal abstract class DiscardRule {
+    abstract fun shouldDiscard(payload: EventPayload): Boolean
+    class All : DiscardRule() {
+        override fun shouldDiscard(payload: EventPayload): Boolean = true
+    }
+
+    class AllHandled : DiscardRule() {
+        override fun shouldDiscard(payload: EventPayload): Boolean = true
```

Comment:
This should be checking the `unhandled` property of the event, not unconditionally discarding all events.